client/components/pie-chart/legend.js

```diff
@@ -0,0 +1,63 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import DataType from './data/type';
+import LegendItem from './legend-item';
+import { sortDataAndAssignSections, isDataEqual } from './data';
+
+class PieChartLegend extends Component {
+	static propTypes = {
+		data: PropTypes.arrayOf( DataType ).isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = this.processData( props.data );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isDataEqual( this.props.data, nextProps.data ) ) {
+			this.setState( this.processData( nextProps.data ) );
+		}
+	}
+
+	processData( data ) {
+		const sortedData = sortDataAndAssignSections( data );
+
+		return {
+			data: sortedData,
+			dataTotal: sortedData.reduce( ( total, datum ) => total + datum.value, 0 ),
+		};
+	}
+
+	render() {
+		const { data, dataTotal } = this.state;
+		return (
+			<div className={ 'pie-chart__legend' }>
+				{ data.map( ( datum, index ) => {
+					return (
+						<LegendItem
+							key={ index.toString() }
```

Comment:
We should preferably use a non-index unique `key`. Very often sometimes like that will be available from the data set we're iterating over. Maybe `datum.name` will work here?

Some background: https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Jeez 😬 yeah, `datum.name` would be a good, unique example.

Reply:
I also need to change this in the `svg` as well.